python/ray/rllib/agents/sac/models.py

```diff
@@ -0,0 +1,179 @@
+from collections import OrderedDict
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from .squash_bijector import SquashBijector
+
+
+SCALE_DIAG_MIN_MAX = (-20, 2)
+
+
+class GaussianLatentSpacePolicy(object):
+    def __init__(self,
+                 input_shape,
+                 output_shape,
+                 hidden_layer_sizes,
+                 squash=True,
+                 activation='relu',
+                 output_activation='linear',
+                 name=None,
+                 *args,
+                 **kwargs):
+        self._squash = squash
+
+        self.condition_inputs = [tf.keras.layers.Input(shape=input_shape)]
+
+        conditions = (
+            tf.keras.layers.Concatenate(axis=-1)(self.condition_inputs)
+            if len(self.condition_inputs) > 1
+            else self.condition_inputs[0])
+
+        out = conditions
+        for units in hidden_layer_sizes:
+            out = tf.keras.layers.Dense(
+                units, *args, activation=activation, **kwargs)(out)
+
+        out = tf.keras.layers.Dense(
+            output_shape[0] * 2, *args,
+            activation=output_activation, **kwargs
+        )(out)
+
+        shift, log_scale_diag = tf.keras.layers.Lambda(
+            lambda shift_and_log_scale_diag: tf.split(
+                shift_and_log_scale_diag,
+                num_or_size_splits=2,
+                axis=-1)
+        )(out)
+
+        log_scale_diag = tf.keras.layers.Lambda(
+            lambda log_scale_diag: tf.clip_by_value(
+                log_scale_diag, *SCALE_DIAG_MIN_MAX)
+        )(log_scale_diag)
+
+        batch_size = tf.keras.layers.Lambda(
+            lambda x: tf.shape(x)[0])(conditions)
+
+        base_distribution = tfp.distributions.MultivariateNormalDiag(
```

Comment:
This should use ModelCatalog.get_action_dist() so that it works with tuple action spaces as well. Btw, are any other changes needed to handle categorical action spaces?